### DISCUSSION TITLE
Idiomatic packaged app cache location

### DISCUSSION THREAD

**Discussion by edomora97** [Homebrew/discussions#3509] [Writing Formulae/Casks] [answered]
Hello everyone!

We are trying to package an application for internal use (we don't plan to publish the Formula to the official repo). This application needs to store cache files somewhere. At the moment these files are version specific (i.e. if there's a version mismatch they are automatically deleted), but this behavior is not permanent: it will be changed in future and eventually the cache will be stable across versions. The cache may contain sensitive information that should not be shared between users. Under Linux the cache is stored inside `$XDG_CACHE_HOME` (i.e. `~/.cache/XXX/`).

1. What should be the _correct_ location for this kind of cache under Homebrew? Should we stick to `$XDG_CACHE_HOME` (i.e. `~/Library/Cache/XXX`) or store it inside its Keg?

2. I'm very new to Homebrew: if user A installs our Formula, can user B use it? If yes, the installed app should not "hardcode" the path to a common cache directory.

Of course since this is internal we don't have to strictly follow guidelines, but we weren't able to find enough resources about that. The best we found is https://docs.brew.sh/Homebrew-and-Python#why-here but I don't think it strictly applies (the cache is created at app's launch, not by the Formula).

Thanks and have a great day!

**Comment by SMillerDev** [comment]
> What should be the correct location for this kind of cache under Homebrew? Should we stick to $XDG_CACHE_HOME (i.e. ~/Library/Cache/XXX) or store it inside its Keg?

Homebrew follows all the standard rules about packaging. It just does so in a prefix for various reasons. What your program uses as file locations is up to you.

> I'm very new to Homebrew: if user A installs our Formula, can user B use it? If yes, the installed app should not "hardcode" the path to a common cache directory.

We don't actively support multi-user usage of Homebrew, but if user B is an admin there is nothing preventing them from accessing files from User A. Including Homebrew.

**Comment by carlocab** [comment, marked as answer]
> 1. What should be the _correct_ location for this kind of cache under Homebrew? Should we stick to `$XDG_CACHE_HOME` (i.e. `~/Library/Cache/XXX`) or store it inside its Keg?

Your cache is created at app launch, so there aren't really any guidelines other than: don't put it in the keg. The keg is not a persistent data store and can change at any time.

> 2. I'm very new to Homebrew: if user A installs our Formula, can user B use it? If yes, the installed app should not "hardcode" the path to a common cache directory.

Typically multiple users will be able to use installed formulae. Not all users may be able to use `brew install`, though.
